components/script/dom/window.rs

```diff
@@ -1540,9 +1555,7 @@ impl Window {
         // When all these conditions are met, notify the constellation
         // that this pipeline is ready to write the image (from the script thread
         // perspective at least).
-        if (opts::get().output_file.is_some() ||
-            opts::get().exit_after_load ||
-            opts::get().webdriver_port.is_some()) &&
+        if (self.output_file_is_some || self.exit_after_load || self.webdriver_port_is_some) &&
```

Comment:
We can simplify this now that we're not relying on specific options - let's store a single boolean value named `prepare_screenshot`, and it can be derived from the appropriate options in pipeline.rs.